scripts/compareAPIs.js

```diff
@@ -2,360 +2,545 @@
+    if (argv.isCI && result.length > 0) {
+      joinedResult = `\`\`\`diff
+${joinedResult}
+\`\`\``;
+    }
+    diffs.push({iname, result: joinedResult, simplifiedName});
+  });
+
+  return {diffs, name};
 }
 
-function generateMessages(matches) {
-  let summaryMessages = [];
+// mirrors dev/docs/src/types.js for the most part
+// "rendering" our types to a string instead of React components
+function processType(value) {
+  if (!value) {
+    console.trace('UNTYPED', value);
+    return 'UNTYPED';
+  }
+  if (Object.keys(value).length === 0) {
+    return '{}';
+  }
+  if (value.type === 'any') {
+    return 'any';
+  }
+  if (value.type === 'null') {
+    return 'null';
+  }
+  if (value.type === 'undefined') {
+    return 'undefined';
+  }
+  if (value.type === 'void') {
+    return 'void';
+  }
+  if (value.type === 'unknown') {
+    return 'unknown';
+  }
+  if (value.type === 'never') {
+    return 'never';
+  }
+  if (value.type === 'this') {
+    return 'this';
+  }
+  if (value.type === 'symbol') {
+    return 'symbol';
+  }
+  if (value.type === 'identifier') {
+    return value.name;
+  }
+  if (value.type === 'string') {
+    if (value.value) {
+      return `'${value.value}'`;
+    }
+    return 'string';
+  }
+  if (value.type === 'number') {
+    return 'number';
+  }
+  if (value.type === 'boolean') {
+    return 'boolean';
+  }
+  if (value.type === 'union') {
+    return value.elements.map(processType).join(' | ');
+  }
+  if (value.type === 'intersection') {
+    return `(${value.types.map(processType).join(' & ')})`;
+  }
+  if (value.type === 'application') {
+    let name = value.base.name;
+    if (!name) {
+      name = processType(value.base);
+    }
+    return `${name}<${value.typeParameters.map(processType).join(', ')}>`;
+  }
+  if (value.type === 'typeOperator') {
+    return `${value.operator} ${processType(value.value)}`;
+  }
+  if (value.type === 'function') {
+    return `(${value.parameters.map(processType).join(', ')}) => ${value.return ? processType(value.return) : 'void'}`;
+  }
+  if (value.type === 'parameter') {
+    return processType(value.value);
+  }
+  if (value.type === 'link') {
+    let name = value.id.substr(value.id.lastIndexOf(':') + 1);
+    if (dependantOnLinks.has(currentlyProcessing)) {
+      let foo = dependantOnLinks.get(currentlyProcessing);
+      if (!foo.includes(name)) {
+        foo.push(name);
+      }
+    } else {
+      dependantOnLinks.set(currentlyProcessing, [name]);
+    }
+    return name;
+  }
+  // interface still needed if we have it at top level?
+  if (value.type === 'object') {
+    if (value.properties) {
+      return `${value.exact ? '{\\' : '{'}
+  ${Object.values(value.properties).map(property => {
+    depth += 2;
+    let result = ' '.repeat(depth);
+    result = `${result}${property.indexType ? '[' : ''}${property.name}${property.indexType ? `: ${processType(property.indexType)}]` : ''}${property.optional ? '?' : ''}: ${processType(property.value)}`;
```

Comment:
I keep looking at all the string outputs like this and wish we were using templates or react style components in some way to make this look cleaner.

Reply:
could definitely be cleaned up, we can do that in followup